cst/plugins/eu.esdihumboldt.cst.functions.collector/src/eu/esdihumboldt/cst/functions/collector/AssignReferenceFromCollector.java

```diff
@@ -0,0 +1,136 @@
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.common.schema.model.constraint.property.Reference;
+import eu.esdihumboldt.hale.common.schema.model.constraint.type.HasValueFlag;
+
+/**
+ * Function to assign references collected by a {@link Collector} to a
+ * multi-valued target property. In case the target property cannot take the
+ * values itself a child property is looked up to assign the references to.
+ * 
+ * @author Florian Esser
+ */
+public class AssignReferenceFromCollector
+		extends AbstractSingleTargetPropertyTransformation<TransformationEngine>
+		implements AssignReferenceFromCollectorFunction {
+
+	private final CollectorGroovyHelper helper = new CollectorGroovyHelper();
+
+	/**
+	 * @see eu.esdihumboldt.hale.common.align.transformation.function.impl.AbstractSingleTargetPropertyTransformation#evaluate(java.lang.String,
+	 *      eu.esdihumboldt.hale.common.align.transformation.engine.TransformationEngine,
+	 *      com.google.common.collect.ListMultimap, java.lang.String,
+	 *      eu.esdihumboldt.hale.common.align.model.impl.PropertyEntityDefinition,
+	 *      java.util.Map,
+	 *      eu.esdihumboldt.hale.common.align.transformation.report.TransformationLog)
+	 */
+	@Override
+	protected Object evaluate(String transformationIdentifier, TransformationEngine engine,
+			ListMultimap<String, PropertyValue> variables, String resultName,
+			PropertyEntityDefinition resultProperty, Map<String, String> executionParameters,
+			TransformationLog log) throws TransformationException, NoResultException {
+
+		// XXX check anchor?
+
+		// assign the value supplied as parameter
+		// conversion will be applied automatically to fit the binding
+//		return getTransformedParameterChecked(PARAMETER_VALUE);
+
+		Collector mainCollector = (Collector) getExecutionContext().getTransformationContext()
+				.get(ContextHelpers.KEY_COLLECTOR);
+		if (mainCollector == null) {
+			throw new TransformationException(
+					"Fatal: No collector has been created yet. Check function priority.");
+		}
+
+		ParameterValue collectorName = getParameterChecked(PARAMETER_COLLECTOR);
+		if (collectorName == null || collectorName.isEmpty()) {
+			throw new TransformationException("Fatal: No collector name was specified.");
+		}
+
+		Collector collector = mainCollector.getAt(collectorName.getValue().toString());
+		if (collector == null) {
+			throw new TransformationException(MessageFormat.format(
+					"Collector \"{0}\" could not (yet) be found. Check spelling and function priority.",
+					collectorName.getValue().toString()));
+		}
+
+		MultiValue result;
+
+		List<Object> collectedReferences = helper.extractCollectedValues(collector);
+		final TypeDefinition resultPropertyType = resultProperty.getDefinition().getPropertyType();
+		if (resultPropertyType.getConstraint(HasValueFlag.class).isEnabled()) {
+			// Target property can take values, assign directly
+			result = new MultiValue(collectedReferences);
```

Comment:
In addition, in case the target element is a reference, we could do the reference processing on the IDs (e.g. when the target is the `href` attribute directly).